### ISSUE TITLE
WSDL derived datatypes in PHPDoc block

### ISSUE THREAD

**Issue by joaorbrandao** [viewflex/zoap#4]
Hi,

So, I am trying to test the Demo code by introducing different WSDL data types (derived datatypes) and i'm getting the following error:
**"Cannot add a complex type anyURI that is not an object or where class could not be found in "DefaultComplexType" strategy."**

This happens when I add a new parameter to getProduct like "@param anyURI $test". How can I use WSDL derived datatypes? Using primitive datatypes works fine...

Many thanks in advance! :)

**Comment by viewflex** [comment]
I think using the fully qualified class name should do the trick.

**Comment by joaorbrandao** [comment]
@viewflex I've got no .../anyURI class. "string" is a native datatype for both WSDL and PHP but "anyURI" is only a native datatype for WSDL, there is no native tive anyURI for PHP. So, in order to use a WSDL anyURI type should I create a PHP class with that name?

**Comment by viewflex** [comment]
No, there is no anyURI type in PHP, but the WSDL anyURI type is essentially just a string. You could create a PHP class to define the type, but I'm not sure that's necessary.

**Comment by joaorbrandao** [comment]
Yes, I think that can be applied for "token" datatype for example @viewflex thanks! But not sure for the "anyURI"... I've got no problem changing anyURI to string, so I think that should work.
There's this image at [w3.org](http://www.w3.org/TR/xmlschema-2/#built-in-datatypes) that shows it.

I'm just getting the same problem when I use a "string[]" type. Should I change the strategy at zoap.php or change the "string[]" to "array"?

**Comment by viewflex** [comment]
Changing to array might work. If you feel like sharing your code I maybe I or someone could suggest a solution.

**Comment by joaorbrandao** [comment]
I followed your previous answer and everything went well @viewflex ! So for the record:

- Fully qualified class name (full path to class) is required.
- WSDL datatypes that in practice are text (token, anyURI, language, etc), can be defined as _string_.
- For the arrays (_string[]_, _SomeComplexType[]_) , I was able to make it work by changing the strategy to _ArrayOfTypeSequence_.

Working with these changes, I was able to use both objects and arrays with simple and complex types.
Everything is working fine! :)
